src/main/java/com/anaplan/client/ServerFile.java

```diff
@@ -311,10 +311,7 @@ public void upLoad(File source, boolean deleteExisting, int chunkSize) throws IO
                 sourceFile.readFully(buffer, 0, size);
                 //reading the last index of the separator
                 int separatorLastIndex = lastIndexOf(buffer, data.getSeparator());
-                // Throw an exception if we don't find a separator in the first chunk
-                if ((totalReadSoFar == 0) && (separatorLastIndex == -1)) {
-                    throw new ParseCSVError(source.getAbsolutePath(), data.getSeparator());
-                }
+                // Throw an exception if we don't find a separator in the first gichunk
```

Comment:
shouldn't this be moved as well?

Reply:
I have removed that code. Now instead of checking seperatorLastIndex, we are checking buffer size.